PIPS/pip-2.md

```diff
@@ -0,0 +1,117 @@
+---
+pip: 2
+title: Relayer Fees
+status: Draft
+author: John Wong (@jwheur)
+created: 2020-06-30
+---
+
+## Simple Summary
+The current LockProxy implementation does not have an in-built method for relayers to charge fees for their work.
+
+This PIP proposes a simple method to charge fees.
+
+## Motivation
+Network fees can be very expensive. For example, the total Ethereum network fees for some popular DApps can cost up to 500k USD over 30 days (source: https://ethgasstation.info/).
+
+## Specification
+
+Example fee deduction implementation for lock:
+```
+struct TxArgs {
+  bytes fromAssetHash;
+  bytes toAssetHash;
+  bytes toAddress;
+  uint256 amount;
+  uint256 feeAmount;
+  bytes feeReceiverAddr;
+}
+
+function lock(
+  address fromAssetHash,
+  uint64 toChainId,
+  bytes memory targetProxyHash,
+  bytes memory toAssetHash,
+  bytes memory toAddress,
+  uint256 amount,
+  bool deductFeeInLock,
+  uint256 feeAmount,
+  bytes memory feeReceiverAddr
+)
+  public
+{
+  bytes32 key = hash(fromAssetHash, toChainId, targetProxyHash, toAssetHash);
+  require(registry[key] == true);
+
+  // Use SafeMath to ensure balances do not overflow
+  balances[key] = balances[key].add(amount);
```

Comment:
balances[key] increment should be `amount - feeAmount` if `deductFeeInLock` equals true.

Reply:
you're right, will change it